replication/row_event.go

```diff
@@ -955,8 +960,12 @@ func (e *RowsEvent) DecodeHeader(data []byte) (int, error) {
 	if e.Version == 2 {
 		dataLen := binary.LittleEndian.Uint16(data[pos:])
 		pos += 2
-
-		e.ExtraData = data[pos : pos+int(dataLen-2)]
+		if dataLen > 2 {
+			err := e.decodeExtraData(pos, data)
```

Comment:
(not important, just a code style) this project more often changes the start of slice at caller, rather than passes the index to function. In other words, I prefer `e.decodeExtraData(data[pos:])`